### PR TITLE
Streamline Build Images workflow using new GitHub Actions features

### PR DESCRIPTION
**Pull request by ashb**
Use `pull_request_target` event for building images, and `concurrency` to
automatically cancel old jobs for PRs.

This means that:

- GitHub will automatically cancel old jobs for us, so we don't have to
  handle that ourselves (removes most of the use of the
  cancel-workflow-action)

  New pushes show this for a little while

  

  And then the old job shows this as the reason

  image

- GitHub displays these checks directly on the PR, but it is still run
  in the context of our repo, meaning it has access write to our
  repo/access to secrets etc.

  

- Since it shows up directly on the PR checks, we don't need to create the
  check in the "CI" workflow to show the status of the Image Build.

- We also don't need to post the comment saying _why_ it failed, as the
  Build Image status will show up directly there

- Since `pull_request_target` has information about the PR in the
  `github.event` context, we don't need the complex mechanism to find
  the "other" PR, we can do a fairly simple API request and filter by
  the commit SHA to find and cancel to CI workflow job. (This removes
  the final use of the cancel-workflow-action)

One change I had to make here what tag we use for Docker images we build
and push up. Previously we used the "source run ID" (i.e. the id of the
CI run) but with pull_request_target we don't have that anymore. We
could use the same API mechanism we do to cancel to find the target job,
but the only requirement here is for an ID that both jobs know -- the
SHA of the PR branch fills that need

Extra side benefits of this:

- The sidebar of commits in main branch aren't "polluted" with Build
  Images for PRs like they were previously.

Still do do:

- [x] ~Update breeze to pull the new image tag format when given a numeric id (use API to look up sha that a build is for)~ Update docs to remove references to RUN_ID
- [x] Delete old build-image-workflow-run workflow file

### CODE REVIEW

**Comment by ashb** [comment]
I've been testing this out on my fork here https://github.com/ashb/airflow/pull/9 -- that is a PR opened from a different user.

**Review by potiuk — COMMENTED** [review]
COOL. :heart: It. The `concurrency` removes major pain out with the cancelling and having it all in one PR makes it cool. 

I love to retire the total madness of the complexity of cancelling because of lack of support from GitHub Actions so far :)

Good stuff @ashb !

**Comment by potiuk** [comment]
> Update breeze to pull the new image tag format when given a numeric id (use API to look up sha that a build is for)

Ths might be not needed. We just need to print SHA in Breeze in instructions and we can use SHA directly. No need to use build ID.

File: .github/workflows/build-images-workflow-run.yml [hunk 1]
```diff
@@ -1,561 +0,0 @@
...
-        uses: ./.github/actions/cancel-workflow-runs
-        id: cancel-failed
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          cancelMode: failedJobs
-          sourceRunId: ${{ github.event.workflow_run.id }}
-          notifyPRCancel: true
-          jobNameRegexps: >
-            ["^Pylint$", "^Static checks", "^Build docs$", "^Spell check docs$",
-             "^Provider packages", "^Checks: Helm tests$", "^Test OpenAPI*"]
```
Comment (ashb) [inline .github/workflows/build-images-workflow-run.yml:124]: Oh, this step is still probably needed, meaning we can't remove the cancel-workflow_runs action entirely.

Comment (potiuk) [reply .github/workflows/build-images-workflow-run.yml:124]: It's an optimisation only, and I am not sure if thats' so much needed after we sped up most of the builds.

Comment (potiuk) [reply .github/workflows/build-images-workflow-run.yml:124]: Selective checks already take care about most savings we have.

**Comment by potiuk** [comment]
Happy to merge this one even without the "fast-fail" for the names of the jobs. I have already put the "deprecation notice" in my action : https://github.com/potiuk/cancel-workflow-runs#readme after the `concurrency` feature added by GitHub Actions.. 

One less thing to maintain :)

**Comment by ashb** [comment]
@potiuk Cool, can you give this a ✅  then please?

And then once merged lets see if we need every open PR to rebase to pick it up or not :/

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
I think no rebase is neeed. It should **just work** .  The name of images did not change, and this is the only interface between the workflows.

**Comment by ashb** [comment]
🤞🏻

**Pull request merged**